src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs

```diff
@@ -124,15 +124,19 @@ public static class WebApiServiceCollectionExtensions
             {
                 options.Events ??= new JwtBearerEvents();
 
+                var onTokenValidatedHandler = options.Events.OnTokenValidated;
+
                 options.Events.OnTokenValidated = async context =>
                 {
                     context.HttpContext.StoreTokenUsedToCallWebAPI(context.SecurityToken as JwtSecurityToken);
                     context.Success();
                     await Task.FromResult(0).ConfigureAwait(false);
+
+                    await onTokenValidatedHandler(context).ConfigureAwait(false);
```

Comment:
```suggestion
                options.Events.OnTokenValidated = async context =>
                {
                    await onTokenValidatedHandler(context).ConfigureAwait(false);                
                    context.HttpContext.StoreTokenUsedToCallWebAPI(context.SecurityToken as JwtSecurityToken);
                    context.Success();
                    await Task.FromResult(0).ConfigureAwait(false);
```

Reply:
Similar to my comments below I would say that in other examples in the library it seems conventional to "do your own stuff THEN call the other event handler".  Is there a specific reason to do differently in this case?  

Also I suspect the  await Task.FromResult(0).ConfigureAwait(false); was only there to stop Resharper/StyleCop/Rosyln nags about async method with no awaits.  There's an "real" await now so that line can go.

Reply:
We want it as @jmprieur suggests because if the event does additional token validation, you don't want to store the token if this validation fails.